lib/auth/github_test.go

```diff
@@ -281,3 +285,73 @@ func TestCalculateGithubUserNoTeams(t *testing.T) {
 	}, &types.GithubAuthRequest{})
 	require.ErrorIs(t, err, ErrGithubNoTeams)
 }
+
+type mockHTTPRequester struct {
+	succeed    bool
+	statusCode int
+}
+
+func (m mockHTTPRequester) Do(req *http.Request) (*http.Response, error) {
+	if !m.succeed {
+		return nil, &url.Error{
+			URL: req.URL.String(),
+			Err: &net.DNSError{
+				IsTimeout: true,
+			},
+		}
+	}
+
+	resp := new(http.Response)
+	resp.Body = io.NopCloser(nil)
+	resp.StatusCode = m.statusCode
+
+	return resp, nil
+}
+
+func TestCheckGithubFeatureSupport(t *testing.T) {
+	connector, err := types.NewGithubConnector("github", types.GithubConnectorSpecV3{
+		TeamsToRoles: []types.TeamRolesMapping{
+			{
+				Organization: "org",
+				Team:         "team",
+				Roles:        []string{"role"},
+			},
+		},
+	})
+	require.NoError(t, err)
+	client := mockHTTPRequester{
+		succeed:    true,
+		statusCode: 200,
+	}
+
+	// Test org having external SSO
```

Comment:
1. Can you split that test, basically create a test table, and run each test case inside `t.Run()`?
2. Can you add a test that covers the re-try/network failure scenario?

Reply:
Sure, split everything into subtests. I already had a test that covered the network failure scenario fyi, just made it into a test case.